### PR TITLE
Fix tldrecho errors

### PR DESCRIPTION
**Pull request by PrinterElf**
L19: get_config_option was not returning anything resulting in config/tldr_echo always being reset to 1
L42-46: If setting is global, remove memory/conv_id/tldr_echo setting so that the rest of the code works.

### CODE REVIEW

File: hangupsbot/plugins/tldr.py [hunk 1]
```diff
@@ -35,9 +35,16 @@ def tldrecho(bot, event, *args):
     else:
         # No path was found. Is this your first setup?
         new_tldr = 0
+    
+    if new_tldr != 2:
```
Comment (nylonee) [inline hangupsbot/plugins/tldr.py]: use `tldr_echo_options[new_tldr] is not "GLOBAL"` instead

Comment (PrinterElf) [inline hangupsbot/plugins/tldr.py]: Done.

File: hangupsbot/plugins/tldr.py [hunk 2]
```diff
@@ -35,9 +35,16 @@ def tldrecho(bot, event, *args):
...
         new_tldr = 0
+    
+    if tldr_echo_options[new_tldr] is not "GLOBAL":
+        # Update the tldr_echo setting
+        bot.memory.set_by_path(['conversations', event.conv_id, 'tldr_echo'], new_tldr)
+    else:
+        # If setting is global then clear the conversation memory entry
+        conv_settings = bot.memory.get_by_path(['conversations', event.conv_id])
+        del conv_settings['tldr_echo'] # remove setting
+        bot.memory.set_by_path(['conversations', event.conv_id], conv_settings)
```
Comment (nylonee) [inline hangupsbot/plugins/tldr.py]: In the current setup, removing `tldr_echo` will force the bot to set it to 0 on the next toggle, which works in this scenario, but the logic isn't ideal. Where in the code is it necessary that the tldr_echo needs to be cleared? Perhaps we can take a look at those parts and see if we can improve that without breaking the tldr_echo toggle logic.

Comment (PrinterElf) [inline hangupsbot/plugins/tldr.py]: That's fine - 0 is the next iteration anyway, and removing it means there's no memory entry so it defaults back to there global setting in config, which is what was being ignored.

Either you put in a special case in the code setting the value to delete it from memory when new_tldr == 2, or you have even messier code handling the fact that there are two situations in which the global setting needs to be used.

For ease of coding, and common sense in terms of interpreting the behaviour, littering memory.json with entries which just tell the code to do what is set in config seems less clean than just deleting the memory entry and relying on the default.

memory.json should (I thought) be containing changes from the default config setting in this instance, not reiterating them.

Comment (PrinterElf) [inline hangupsbot/plugins/tldr.py]: Re-reading this, I should point out that the previous code had no mechanism for the second of the two pseudocode examples above as the method was to remove the memory entry and revert to using config default.

Comment (PrinterElf) [inline hangupsbot/plugins/tldr.py]: That's fine - 0 is the next iteration anyway, and removing it means there's no memory entry so it defaults back to the global setting in config, which is what was being ignored.

Either you put in a special case in the code setting the value to delete it from memory when new_tldr == 2, or you have even messier code handling the fact that there are two situations in which the GLOBAL enumeration needs to be parsed.

is memory set? no? Use config.
OR
is memory set? yes? Ok, use that value.
Oh, that value is global, then I need to go back and get the config value again anyway.

For ease of coding, and common sense in terms of interpreting the behaviour, littering memory.json with entries which just tell the code to do what is set in config seems less clean than just deleting the memory entry and relying on the default.

memory.json should (I thought) be containing changes from the default config setting in this instance, not reiterating them.

Comment (PrinterElf) [inline hangupsbot/plugins/tldr.py]: Re-reading this, I should point out that the previous code had no mechanism for the second of the two pseudocode examples above as the method was to remove the memory entry and revert to using config default.

**Pull request merged**